sdk/src/Services/DynamoDBv2/Custom/DocumentModel/TableBuilder.cs

```diff
@@ -0,0 +1,260 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+using Amazon.DynamoDBv2.Model;
+using System;
+using System.Collections.Generic;
+
+namespace Amazon.DynamoDBv2.DocumentModel
+{
+    /// <summary>
+    /// Builder that constructs a <see cref="Table"/>
+    /// </summary>
+    public class TableBuilder : ITableBuilder
+    {
+        /// <summary>
+        /// The <see cref="Table"/> object that is built and then returned from <see cref="Build"/>
+        /// </summary>
+        private Table _table;
+
+        /// <summary>
+        /// Keeps track internally of attributes that have already been saved in <see cref="Table.Attributes"/>,
+        /// since they can be shared by different indices.
+        /// </summary>
+        private HashSet<string> _attributesAlreadyProcessed; 
+
+        /// <summary>
+        /// Creates a builder object to construct a <see cref="Table"/>
+        /// </summary>
+        /// <param name="ddbClient">Client to use to access DynamoDB.</param>
+        /// <param name="tableName">Table name</param>
+        public TableBuilder(IAmazonDynamoDB ddbClient, string tableName) :
+             this(ddbClient, tableName, DynamoDBEntryConversion.CurrentConversion, false, null)
+        {
+        }
+
+        /// <summary>
+        /// Creates a builder object to construct a <see cref="Table"/>
+        /// </summary>
+        /// <param name="ddbClient">Client to use to access DynamoDB.</param>
+        /// <param name="tableName">Table name</param>
+        /// <param name="conversion">Conversion to use for converting .NET values to DynamoDB values.</param>
+        /// <param name="isEmptyStringValueEnabled">If the property is false, empty string values will be interpreted as null values.</param>
+        /// <param name="metadataCachingMode">The document API relies on an internal cache of the DynamoDB table's metadata to construct and validate 
+        /// requests. This controls how the cache key is derived, which influences when the SDK will call 
+        /// <see cref="IAmazonDynamoDB.DescribeTable(string)"/> internally to populate the cache.</param>
+        public TableBuilder(IAmazonDynamoDB ddbClient, string tableName, DynamoDBEntryConversion conversion, bool isEmptyStringValueEnabled, MetadataCachingMode? metadataCachingMode)
```

Comment:
Why have this constructor since the next one that takes in a `TableConfig` covers this use case?

Reply:
I was matching our most-explicit constructor for `Table` here: https://github.com/aws/aws-sdk-net/blob/7cba18c9af7d48bc307b7f290e7a7993cbec2dd4/sdk/src/Services/DynamoDBv2/Custom/DocumentModel/Table.cs#L508

The catch with only taking in a `TableConfig` is that we don't expose `IsEmptyStringValueEnabled`
https://github.com/aws/aws-sdk-net/blob/7cba18c9af7d48bc307b7f290e7a7993cbec2dd4/sdk/src/Services/DynamoDBv2/Custom/DocumentModel/TableConfig.cs#L57

So the only way to set that is through the `Table` constructor that builds a `TableConfig` using our internal constructor.

Not entirely sure why this is like this. I agree a constructor that takes `TableConfig` is more maintainable since we don't have to keep adding new arguments. Should we just flip `TableConfig.IsEmptyStringvalueEnabled` to `public`?

Reply:
Possibly because we do want to make sure we set `TableConfig.Consumer` accurately, we construct `TableConfig` on behalf of the user?

Reply:
I'm good with leaving as is. Makes sense to match the `LoadTable` overloads.